lib/kubernetes-deploy/local_resource_discovery.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module KubernetesDeploy
+  class LocalResourceDiscovery
+    def initialize(template_dirs:, namespace:, context:, current_sha:, logger:, bindings:, namespace_tags:, crds: {})
+      @template_dirs = template_dirs
+      @namespace = namespace
+      @context = context
+      @logger = logger
+      @namespace_tags = namespace_tags
+      @renderers = Hash.new do |hash, template_dir|
+        hash[template_dir] = KubernetesDeploy::Renderer.new(
+          current_sha: current_sha,
+          template_dir: template_dir,
+          logger: logger,
+          bindings: bindings,
+        )
+      end
+      @crds = crds
+    end
+
+    def resources
+      resources = []
+      templates.each do |template_dir, filenames|
+        filenames.each do |filename|
+          split_templates(template_dir, filename) do |r_def|
```

Comment:
Thinking about implementing `-f` (e.g. include _filenames_, not just _template_dirs_):

I think we'll be able to do it here, at least in the context of `DeployTask`. We can process directory arguments as we currently do, but for the remaining filenames, we can gather them up by directory and create a similiar `hash[template_dir]: [ONLY_SPECIFIED_FILES]` and proceed as normal.

I'm wondering if I'm painting myself into a corner where I'll have to reimplement `-f` differently for `krane render`, which wouldn't be the greatest (since this class is intrinsically linked with _resource_ discovery, which is not a concern of the render task).

In fact, `krane render` couldn't even instantiate a `LocalResourceDiscovery` object without passing in a ton of `nil` fields (to say nothing of it being semantic nonsense) :( 

Maybe what we need to actually do is move the creation of `template_dir -> filenames` mappings further up the stack instead of in `LocalResourceDiscovery#templates`. I don't want to write the `-f` filter in multiple places